tests/acceptance/features/apiSharingNgLinkShare/linkShare.feature

```diff
@@ -2971,3 +2971,74 @@ Feature: Create a share link for a resource
       | createOnly       | view                 |
       | createOnly       | edit                 |
       | createOnly       | upload               |
+
+
+  Scenario Outline: update link share of a file inside a project-space using permissions endpoint
+    Given using spaces DAV path
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "projectSpace" with the default quota using the Graph API
+    And user "Alice" has uploaded a file inside space "projectSpace" with content "to share" to "textfile.txt"
+    And user "Alice" has created the following resource link share:
+      | resource        | textfile.txt       |
+      | space           | projectSpace       |
+      | permissionsRole | <permissions-role> |
+      | displayName     | Link               |
+      | password        | %public%           |
+    When user "Alice" updates the last public link share using the permissions endpoint of the Graph API:
+      | resource           | textfile.txt           |
+      | space              | projectSpace           |
+      | permissionsRole    | <new-permissions-role> |
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should match
+      """
+        {
+          "type": "object",
+          "required": [
+            "hasPassword",
+            "id",
+            "link",
+            "createdDateTime"
+          ],
+          "properties": {
+            "hasPassword": {
+              "const": true
+            },
+            "id": {
+              "type": "string",
+              "pattern": "^[a-zA-Z]{15}$"
+            },
+            "link": {
+              "type": "object",
+              "required": [
+                "@libre.graph.displayName",
+                "@libre.graph.quickLink",
+                "preventsDownload",
+                "type",
+                "webUrl"
+              ],
+              "properties": {
+                "@libre.graph.displayName": {
+                  "const": "Link"
+                },
+                "@libre.graph.quickLink": {
+                  "const": false
+                },
+                "preventsDownload": {
+                  "const": false
+                },
+                "type": {
+                  "const": "<new-permissions-role>"
+                },
+                "webUrl": {
+                  "type": "string",
+                  "pattern": "^%base_url%/s/[a-zA-Z]{15}$"
+                }
+              }
+            }
+          }
+        }
+        """
+    Examples:
+      | permissions-role | new-permissions-role |
+      | view             | edit                 |
+      | edit             | view                 |
```

Comment:
any specific reason for limiting permissions-role to vie and edit only?

Reply:
It was due to different JSON response for edit/view and internal permissions role. I have added new scenario for updating role of a file to internal.